src/rgw/rgw_rados.cc

```diff
@@ -5930,10 +5930,14 @@ int RGWRados::append_atomic_test(RGWObjectCtx *rctx,
   if (r < 0)
     return r;
 
-  RGWObjState *state = *pstate;
+  return append_atomic_test(*pstate, op);
```

Comment:
dont think I remember what atomic_test does

Reply:
the initial head object read happens in `RGWRados::get_obj_state()`, which prefetches the first chunk of data along with all the attributes. after that, any additional reads from the head object add this 'atomic check' which uses op.cmpxattr() to make sure we're reading from the same object

i don't think really happens in practice though, because AtomicObjectProcessor will only write a single chunk to the head object, and Multipart data is only in its part objects